src/plugins/data-card/data-card-tile.tsx

```diff
@@ -55,6 +56,15 @@ export const DataCardToolComponent: React.FC<ITileProps> = observer((props) => {
     });
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    if (!tileElt) return;
+    const uiHeight = tileElt?.querySelector(".data-card-container")?.scrollHeight || 0;
+    const spaceLeft = height ? height - uiHeight : 0;
+    if (readOnly) onRequestRowHeight(model.id, Math.max(uiHeight, kExampleDeckHeight));
```

Comment:
I'm not completely sure, but I think you shouldn't be setting the row height when read only. The row height is saved in the model. So if you are doing this in readOnly it means you are "writting". More concretely if the document is rendered in multiple places (often happens on left side) they will each be trying to update the row height.

Reply:
yes but this issue with cards not showing themselves happens in content on the left side as well. It doesn't do anyone any good if you can't see all the stuff in a card - the moth images were getting sliced off halfway even if they were in the first field. So I think you gotta set heights everywhere no? @scytacki

Reply:
Paired briefly with Scott and implemented so that we allow this operation when we are in content.

Reply:
@lbondaryk I'd note that the height will not be adjusted automatically for non-curriculum documents that are being viewed "readOnly". I'd think this will be OK since the height ought to be adjusted when the user original created or edited the document.